### PR TITLE
Rectangular mtd cleaning

### PR DESCRIPTION
**Pull request by parbol**
#### PR description:

This PR proposes the following main changes:

- A cleaning and simplification of the Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h class and surrounding classes. The previous code was based on the tracker code and contained many unneeded features for the MTD purposes. These changes by themselves should not affect any output as we have tested locally. 

- Cleaning of the MTDTopology class Geometry/MTDNumberingBuilder/interface/MTDTopology.h, which was essentially unused in the code with the exception of a couple of methods. These changes by themselves should not affect any output as we have tested locally.
  
- We have also inverted the order of the 11th and 12th positions of the ETL vector in mtdParameters.xml in such a way that the 32 pixels are aligned with the long side of the sensor, and the 16 pixels with the short side (It was the other way around before). 

- Addition of new information and methods in the RectangularMTDTopology class to include the non-active interpixel gaps for the ETL and the gaps between the pixels and the sensor borders. This 4 parameters (2 for the x coordinate and 2 for the y coordinate) have been added to mtdParameters.xml. The second 4 positions of the ETL parameters vector have been set to 0 as they are not used anywhere else in the code. This change should slightly change the acceptance of the ETL sensors (since small 50 microns gaps should appear in between pixels and between the pixels and the border of the sendor).

- Modifications have been made to Geometry/MTDGeometryBuilder/test/MTDDigiGeometryAnalyzer.cc in order to dump the position of the pixels, and also to SimFastTiming/FastTimingCommon/src/ETLDeviceSim.cc in order to discard simhits landing on the gaps.

#### PR validation:

- We have checked that the only change in the Digitization is related to the gaps in between the ETL pixels. In this link you can see how the simhits look like with slightly exagerated interpad distances (150 microns in the plots vs. the 50 more realistic gap in the code)
 
https://cernbox.cern.ch/index.php/s/xweGn88Zs9XN036

- We have checked that the positions of the pixels in the ETL sensors are as expected.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test with cms-data/Geometry-TestReference#6

File: Geometry/MTDGeometryBuilder/src/MTDGeomBuilderFromGeometricTimingDet.cc [hunk 1]
```diff
@@ -103,14 +103,23 @@ void MTDGeomBuilderFromGeometricTimingDet::buildPixel(
 
   // this is a hack while we put things into the DDD
   int ROCrows(0), ROCcols(0), ROCSx(0), ROCSy(0);
+  int GAPxInterpad(0), GAPxBorder(0), GAPyInterpad(0), GAPyBorder(0);
   switch (det) {
     case GeomDetType::SubDetector::TimingBarrel:
+      GAPxInterpad = ptp.vitems_[0].vpars_[0];
+      GAPxBorder = ptp.vitems_[0].vpars_[1];
+      GAPyInterpad = ptp.vitems_[0].vpars_[2];
+      GAPyBorder = ptp.vitems_[0].vpars_[3];
```
Comment (cvuosalo) [inline Geometry/MTDGeometryBuilder/src/MTDGeomBuilderFromGeometricTimingDet.cc:112]: Please add a comment here documenting that the border and interpad values are in microns (or whatever the correct unit  is).

File: Geometry/MTDGeometryBuilder/interface/MTDTopologyBuilder.h [hunk 2]
```diff
@@ -14,13 +14,14 @@ class MTDTopologyBuilder {
...
-                       int BIG_PIX_PER_ROC_Y,  // in y direction, cols. BIG_PIX_PER_ROC_Y = 0 for SLHC
+                       int ROWS_PER_ROC,  // Num of Rows per ROC
+                       int COLS_PER_ROC,  // Num of Cols per ROC
                        int ROCS_X,
-                       int ROCS_Y);
+                       int ROCS_Y,
+                       int GAPxInterpad,
+                       int GAPxBorder,
+                       int GAPyInterpad,
+                       int GAPyBorder);
```
Comment (cvuosalo) [inline Geometry/MTDGeometryBuilder/interface/MTDTopologyBuilder.h:24]: Please add a comment that the border and interpad values have to be in microns here.

File: Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h [hunk 3]
```diff
@@ -43,13 +43,14 @@ class RectangularMTDTopology final : public PixelTopology {
...
-                         int BIG_PIX_PER_ROC_Y,  // in y direction, cols. BIG_PIX_PER_ROC_Y = 0 for SLHC
+                         int ROWS_PER_ROC,  // Num of Rows per ROC
+                         int COLS_PER_ROC,  // Num of Cols per ROC
                          int ROCS_X,
-                         int ROCS_Y)
+                         int ROCS_Y,
+                         float GAPxInterpad,
+                         float GAPxBorder,
+                         float GAPyInterpad,
+                         float GAPyBorder)
```
Comment (cvuosalo) [inline Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h:53]: Please add a comment here that the border and interpad values have to be in cm in the constructor.

File: Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h [hunk 4]
```diff
@@ -175,6 +159,14 @@ class RectangularMTDTopology final : public PixelTopology {
   int colsperroc() const override { return m_COLS_PER_ROC; }
   float xoffset() const { return m_xoffset; }
   float yoffset() const { return m_yoffset; }
+  float gapxInterpad() const { return m_GAPxInterpad; }
+  float gapyInterpad() const { return m_GAPyInterpad; }
+  float gapxBorder() const { return m_GAPxBorder; }
+  float gapyBorder() const { return m_GAPyBorder; }
```
Comment (cvuosalo) [inline Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h:165]: Please add a comment here that the border and interpad values are returned as cm by these methods.

**Comment by fabiocos** [comment]
@parbol concerning the clang-tidy warning for the digitization, it is likely simpler to solve alla of the in a dedicated separate PR.

**Comment by fabiocos** [comment]
please test with cms-data/Geometry-TestReference#6

**Comment by cvuosalo** [comment]
+1

**Comment by fabiocos** [comment]
@parbol I guess that most of the apparently unrelated differences are linked to the modified use of the same sequence of random numbers in digitizers. Is the output what you expect?

**Comment by civanch** [comment]
+1

**Comment by srimanob** [comment]
+Upgrade

**Comment by fabiocos** [comment]
@silviodonato @qliphy for the integration, this has to go in parallel with https://github.com/cms-data/Geometry-TestReference/pull/6 otherwise one MTD geometry unit test will fail

**Comment by qliphy** [comment]
+1

**Comment by fabiocos** [comment]
@parbol the update of the parameters in ```mtdParameters.xml``` is done only for the new ETL (D76 and following), but the code using them is applied also to the older scenarios D49 and D60, I do not see any protection against this. With the old default parameters, I guess we are getting close to random results, and this should explain the differences observed in #33598 test, do you agree? Setting all the first eight parameters to 0 is the way to preserve the old behaviour I would say.

**Comment by parbol** [comment]
Hi Fabio,

Yes, this makes sense. I will update it.

**Pull request merged**